### PR TITLE
Revamp module build script to make it work for 5.15 on Ubuntu 20.04

### PR DESCRIPTION
**Pull request by saiarcot895**
The current version of the script does not work at all when the host kernel is Azure's 5.15 kernel on Ubuntu 20.04. It does seem to work with 5.11 and older.

Update the script to add support for building and loading the kernel modules for the 5.15 kernel, while keeping support for the non-HWE kernel.

In addition, use a separate `updates/sonic` directory for the kernel modules we compile, to make it clear they were compiled separately and don't come from the official kernel build packages.

Finally, add a warning if we're compiling a kernel module for a different version of the kernel than what is currently running.

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>

### CODE REVIEW

**Comment by saiarcot895** [comment]
@xumia @qiluo-msft could you review this when you get a chance?

File: .azure-pipelines/build_and_install_module.sh [hunk 1]
```diff
@@ -26,64 +26,62 @@ function build_and_install_kmodule()
...
+    fi
     make  allmodconfig
     mv .config .config.bk
     cp /boot/config-$(uname -r) .config
     grep NET_TEAM .config.bk >> .config
-    echo CONFIG_NET_VRF=m >> .config
-    echo CONFIG_MACSEC=m >> .config
-    echo CONFIG_NET_VENDOR_MICROSOFT=y >> .config
-    echo CONFIG_MICROSOFT_MANA=m >> .config
-    echo CONFIG_SYSTEM_REVOCATION_LIST=n >> .config
```
Comment (qiluo-msft) [inline .azure-pipelines/build_and_install_module.sh:55]: why removing?

Comment (saiarcot895) [reply .azure-pipelines/build_and_install_module.sh:55]: Ideally, when building kernel modules, the config used for building the kernel module must match the system's kernel config. This is especially true for config options that aren't for just building a module (i.e. options that may affect multiple parts of the kernel). If there's a difference, there a risk of structure sizes not matching, or function parameters not matching.

These kernel options, as far as I can tell, and based on the CI test run, don't have any functional impact.

Comment (saiarcot895) [reply .azure-pipelines/build_and_install_module.sh:55]: Ideally, when building kernel modules, the config used for building the kernel module must match the system's kernel config. This is especially true for config options that aren't for just building a module (i.e. options that may affect multiple parts of the kernel). If there's a difference, there a risk of structure sizes not matching, or function parameters not matching.

**Comment by saiarcot895** [comment]
@xumia @liushilongbuaa Could you review this when you get a chance?

**Review by xumia — APPROVED** [review]

**Comment by xumia** [comment]
> @xumia @liushilongbuaa Could you review this when you get a chance?

LGTM, thanks for the fix.

**Pull request merged**